src/Integration/Persistence/LegacySolutionBindingPathProvider.cs

```diff
@@ -20,28 +20,31 @@
 
 using System;
 using System.IO;
-using SonarLint.VisualStudio.Core.Binding;
+using Microsoft.VisualStudio.Shell.Interop;
 
 namespace SonarLint.VisualStudio.Integration.Persistence
 {
     internal class LegacySolutionBindingPathProvider : ISolutionBindingPathProvider
     {
-        private readonly IServiceProvider serviceProvider;
-
+        private readonly IVsSolution vsSolution;
         public const string LegacyBindingConfigurationFileName = "SolutionBinding.sqconfig";
 
-        public LegacySolutionBindingPathProvider(IServiceProvider serviceProvider)
-        { 
-            this.serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+        public LegacySolutionBindingPathProvider(IVsSolution vsSolution)
```

Comment:
Prep work to be able to consume this class without being dependent on LocalServices - `ISolutionRuleSetsInformationProvider` was in the way, and the method GetSolutionSonarQubeRulesFolder was used only here. I changed it to work with `IVsSolution` instead of DTE.

Reply:
How are you planning to create this path provider and the other one going forwards? 

`IVsSolution` isn't directly MEF-importable so they will have to be explicitly created by something else. If you make the constructor import `SVSServiceProvider` then they could be imported directly.

Reply:
Yep, that's the plan.
Once this PR and  #1851 are done, I'll new-up dependencies of ConfigurationProvider and to them IServiceProvider.